File: Temperature_Converter.c
#include<stdio.h>

int main() {
    int choice;
    double temp, result;

    printf("=====================================\n");
    printf("      TEMPERATURE CONVERTER\n");
    printf("=====================================\n");

    printf("Choose Conversion:\n");
    printf("1. Celsius to Fahrenheit\n");
    printf("2. Celsius to Kelvin\n");
    printf("3. Fahrenheit to Celsius\n");
    printf("4. Fahrenheit to Kelvin\n");
    printf("5. Kelvin to Celsius\n");
    printf("6. Kelvin to Fahrenheit\n");
    printf("Enter your choice (1-6): ");
    scanf("%d", &choice);

    printf("Enter temperature value: ");
    scanf("%lf", &temp);

    switch (choice) {
        case 1:
            result = (temp * 9.0 / 5.0) + 32;
            printf("%.2lf Celsius = %.2lf Fahrenheit\n", temp, result);
            break;

        case 2:
            result = temp + 273.15;
            printf("%.2lf Celsius = %.2lf Kelvin\n", temp, result);
            break;

        case 3:
            result = (temp - 32) * 5.0 / 9.0;
            printf("%.2lf Fahrenheit = %.2lf Celsius\n", temp, result);
            break;

        case 4:
            result = ((temp - 32) * 5.0 / 9.0) + 273.15;
            printf("%.2lf Fahrenheit = %.2lf Kelvin\n", temp, result);
            break;

        case 5:
            if (temp < 0) {
                printf("Invalid Kelvin temperature! Kelvin cannot be negative.\n");
                return 1;
            }
            result = temp - 273.15;
            printf("%.2lf Kelvin = %.2lf Celsius\n", temp, result);
            break;

        case 6:
            if (temp < 0) {
                printf("Invalid Kelvin temperature! Kelvin cannot be negative.\n");
                return 1;
            }
            result = ((temp - 273.15) * 9.0 / 5.0) + 32;
            printf("%.2lf Kelvin = %.2lf Fahrenheit\n", temp, result);
            break;

        default:
            printf("Invalid choice! Please select between 1 and 6.\n");
    }

    return 0;
}

File: TemperatureConverter.java
import javax.swing.*;
import java.awt.*;
import java.awt.event.*;

public class TemperatureConverter extends JFrame implements ActionListener {

    JLabel titleLabel, tempLabel, scaleLabel, resultLabel;
    JTextField tempField;
    JComboBox<String> scaleBox;
    JButton convertButton;

    public TemperatureConverter() {

        setTitle("Temperature Converter");
        setSize(450, 300);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setLayout(null);

        titleLabel = new JLabel("Temperature Converter");
        titleLabel.setFont(new Font("Arial", Font.BOLD, 20));
        titleLabel.setBounds(110, 20, 250, 30);

        tempLabel = new JLabel("Enter Temperature:");
        tempLabel.setBounds(50, 80, 150, 25);

        tempField = new JTextField();
        tempField.setBounds(200, 80, 150, 25);

        scaleLabel = new JLabel("Select Scale:");
        scaleLabel.setBounds(50, 120, 150, 25);

        String[] scales = {"Celsius", "Fahrenheit", "Kelvin"};
        scaleBox = new JComboBox<>(scales);
        scaleBox.setBounds(200, 120, 150, 25);

        convertButton = new JButton("Convert");
        convertButton.setBounds(150, 170, 120, 35);
        convertButton.addActionListener(this);

        resultLabel = new JLabel("Result will appear here");
        resultLabel.setBounds(50, 220, 350, 25);

        add(titleLabel);
        add(tempLabel);
        add(tempField);
        add(scaleLabel);
        add(scaleBox);
        add(convertButton);
        add(resultLabel);

        setVisible(true);
    }

    @Override
    public void actionPerformed(ActionEvent e) {

        try {
            double temp = Double.parseDouble(tempField.getText());
            String scale = (String) scaleBox.getSelectedItem();

            double celsius, fahrenheit, kelvin;

            if (scale.equals("Celsius")) {
                fahrenheit = (temp * 9 / 5) + 32;
                kelvin = temp + 273.15;

                resultLabel.setText(
                    String.format("Fahrenheit: %.2f °F | Kelvin: %.2f K",
                            fahrenheit, kelvin));

            } else if (scale.equals("Fahrenheit")) {
                celsius = (temp - 32) * 5 / 9;
                kelvin = celsius + 273.15;

                resultLabel.setText(
                    String.format("Celsius: %.2f °C | Kelvin: %.2f K",
                            celsius, kelvin));

            } else {
                celsius = temp - 273.15;
                fahrenheit = (celsius * 9 / 5) + 32;

                resultLabel.setText(
                    String.format("Celsius: %.2f °C | Fahrenheit: %.2f °F",
                            celsius, fahrenheit));
            }

        } catch (NumberFormatException ex) {
            JOptionPane.showMessageDialog(this,
                    "Please enter a valid number!",
                    "Input Error",
                    JOptionPane.ERROR_MESSAGE);
        }
    }

    public static void main(String[] args) {
        new TemperatureConverter();
    }
}